armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1174,18 +1176,44 @@ def buildRingSchedule(
         findAssembly
 
         """
+        maxRingInCore = self.r.core.getNumRings()
+        if dischargeRing > maxRingInCore:
+            runLog.warning(
+                f"Discharge ring {dischargeRing} is outside the core (max {maxRingInCore}). "
+                "Changing it to be the max ring"
+            )
+            dischargeRing = maxRingInCore
+        if chargeRing > maxRingInCore:
+            runLog.warning(
+                f"Charge ring {chargeRing} is outside the core (max {maxRingInCore}). "
+                "Changing it to be the max ring."
+            )
+            chargeRing = maxRingInCore
+
         # process arguments
         if dischargeRing is None:
-            # default to convergent
+            # No discharge ring given, so we default to converging from outside to inside
+            # and therefore discharging from the center
             dischargeRing = 1
         if chargeRing is None:
-            chargeRing = self.r.core.getNumRings()
-
-        if chargeRing > dischargeRing and not jumpRingTo:
+            # Charge ring not specified. Since we default to convergent shuffling, we
+            # must insert the fuel at the periphery.
+            chargeRing = maxRingInCore
+        if jumpRingFrom is not None and not (1 < jumpRingFrom < maxRingInCore):
+            raise ValueError(f"JumpRingFrom {jumpRingFrom} is not in the core.")
+        if jumpRingTo is not None and not (1 < jumpRingTo < maxRingInCore):
+            raise ValueError(f"JumpRingTo {jumpRingTo} is not in the core.")
+
+        if chargeRing > dischargeRing and jumpRingTo is None:
+            # a convergent shuffle with no jumping. By setting
+            # jumpRingTo to be 1, no jumping will be activated
+            # in the later logic.
             jumpRingTo = 1
-        elif not jumpRingTo:
+        elif jumpRingTo is None:
+            # divergent case. Disable jumping by putting jumpring
+            # at periphery.
             if self.r:
-                jumpRingTo = self.r.core.getNumRings()
+                jumpRingTo = maxRingInCore
             else:
                 jumpRingTo = 18
```

Comment:
Sort of an arbitrary choice, right? Can we do better than this now or at least document why the number 18 is selected?

Reply:
Huh. I never even noticed that 18 before.

At the very least, we should probably add a comment for why that magic number was chosen.

But it scares me a little, yeah.

Reply:
@ntouran, can you look into this an explain why 18 is still a good number or think of maybe a better way to handle this. I think this code is pretty old so maybe the framework changes have made it easier to set a better max value?

Reply:
I think this should still be addressed. I'm not sure where `18` comes from, and that makes me nervous.